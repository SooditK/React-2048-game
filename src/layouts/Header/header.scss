@import '../../assets/styles/vars';
$web-height: 5vw;

.header {
  background-color: #3f51b5;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
  color: #fff;
  display: flex;
  flex-direction: column;
  flex-shrink: 0;
  flex-wrap: nowrap;
  font-size: 1.5rem;
  justify-content: flex-start;
  margin: 0;
  max-height: 50vh;
  min-height: $web-height;
  padding: 0;
  transition-duration: 0.2s;
  transition-property: max-height, box-shadow;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  width: 100%;
  z-index: 3;

  a {
    color: #fff;
    text-decoration: none;
  }

  li {
    list-style-type: none;
  }

  .row {
    align-items: center;
    align-self: stretch;
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    flex-wrap: nowrap;
    height: $web-height;
    margin: 0;
    padding: 0 3vw 0 5vw;
  }

  .row > * {
    flex-shrink: 0;
  }

  .title {
    box-sizing: border-box;
    display: block;
    font-weight: 400;
    letter-spacing: 0.02em;
    line-height: 1;
    padding-right: 1.5vw;
    position: relative;
  }

  .spacer {
    flex-grow: 1;
  }

  .link {
    letter-spacing: 0;
    // line-height: 64px;
    // opacity: 0.87;
    padding: 0 1.5vw;
  }

  .active {
    color: #ff4081;
  }

  .icon {
    width: 3vw;
  }
}
$mobile-height: 15vw;

@media #{$mobile} {
  .header {
    font-size: 1.35rem;
    min-height: $mobile-height;

    .row {
      min-height: $mobile-height;
    }

    .icon {
      width: 8vw;
    }
  }
}
