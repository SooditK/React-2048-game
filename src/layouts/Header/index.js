import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './header.scss';
import github from '../../assets/images/github-white.png';

// The Header creates links that used to navigate
// between routes.
export default class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isMobile: window.innerWidth <= 768
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.mobileDetect, false);
  }

  mobileDetect = () => {
    this.setState({
      isMobile: window.innerWidth <= 768
    });
  };

  render() {
    const { isMobile } = this.state;
    return (
      <header className={styles.header}>
        <nav>
          <ul className={styles.row}>
            <li className={styles.title}>
              {/* eslint-disable jsx-a11y/anchor-is-valid */}
              <NavLink exact to="/" activeClassName={styles.active}>
                2048 Game
              </NavLink>
            </li>
            {isMobile ? null : (
              <li className={styles.link}>
                <h3 to="/ranking" activeClassName={styles.active}>
                  Perseverance Team
                </h3>
              </li>
            )}
            <li className={styles.spacer} />
            <li>
              <a
                href="https://github.com/devrsi0n/React-2048-game"
                target="_blank"
                rel="noopener noreferrer"
              >
                <img src={github} alt="github repo" className={styles.icon} />
              </a>
            </li>
          </ul>
        </nav>
      </header>
    );
  }
}
