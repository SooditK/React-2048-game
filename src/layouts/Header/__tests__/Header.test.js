import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Header from '..';

describe('<Header />', () => {
  it('component render', () => {
    const header = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
