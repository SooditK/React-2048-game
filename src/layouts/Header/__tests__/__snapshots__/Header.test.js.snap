// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component render 1`] = `
<header
  className="header"
>
  <nav>
    <ul
      className="row"
    >
      <li
        className="title"
      >
        <a
          aria-current="true"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/ranking"
          onClick={[Function]}
          style={undefined}
        >
          Ranking
        </a>
      </li>
      <li
        className="link"
      >
        <a
          aria-current={false}
          className={undefined}
          href="/comments"
          onClick={[Function]}
          style={undefined}
        >
          Comments
        </a>
      </li>
      <li
        className="spacer"
      />
      <li>
        <a
          href="https://github.com/devrsi0n/React-2048-game"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="github repo"
            className="icon"
            src="github-white.png"
          />
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
